File: pulsar/java/src/test/java/com/examples/hello/pulsar/ApplicationTest.java
package com.examples.hello.pulsar;

import io.micronaut.runtime.EmbeddedApplication;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

@MicronautTest
class ApplicationTest {
  @Inject
  EmbeddedApplication<?> application;

  @Test
  void testItWorks() {
    assertThat(application.isRunning()).isTrue();
  }
}

File: pulsar/kotlin/src/test/kotlin/com/example/hello/pulsar/ApplicationTest.kt
package com.example.hello.pulsar

import io.micronaut.runtime.EmbeddedApplication
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.assertj.core.api.AssertionsForClassTypes
import org.junit.jupiter.api.Test

@MicronautTest
internal class ApplicationTest {
    @Inject
    var application: EmbeddedApplication<*>? = null
    @Test
    fun testItWorks() {
        AssertionsForClassTypes.assertThat(application!!.isRunning).isTrue
    }
}
